register/models.py

```diff
@@ -122,8 +139,14 @@ def send_reimbursement(self, request):
         self.save()
 
     def confirm(self, cancellation_url):
+        if self.status in [APP_ACCEPTED,APP_PENDING,APP_REJECTED]:
+            raise Http404('Unfortunately this invite doesn\'t even exist')
```

Comment:
Agree, also, the condition if its more general than the other cases could be the last one and just be if different than invited, Right? Would that make more sense? Not important though